File: questions/1656. Design an Ordered Stream/design.cpp
class OrderedStream {
vector<string> arr;
int ptr = 1;  
public:
  
    OrderedStream(int n): arr(n + 1){}
    vector<string> insert(int id, string value) {
        vector<string> res;
        arr[id] = value;
        if(id == ptr){
            while(ptr < arr.size() && !arr[ptr].empty()){
                res.push_back(arr[ptr]);
                ++ptr;
            }
        }
        return res;
    }
};

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream* obj = new OrderedStream(n);
 * vector<string> param_1 = obj->insert(id,value);
 */

File: questions/1656. Design an Ordered Stream/design.java
class OrderedStream {
    String[] arr;
    int ptr;
    public OrderedStream(int n) {
        arr = new String[n];
        ptr = 0;
    }
    
    public List<String> insert(int id, String value) {
        List<String> res = new ArrayList();
        arr[id - 1] = value;
        if(ptr == id - 1){
            while(ptr < arr.length && arr[ptr] != null){
                res.add(arr[ptr++]);
            }
        }
        return res;
    }
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream obj = new OrderedStream(n);
 * List<String> param_1 = obj.insert(id,value);
 */
